### PR TITLE
Disable external entities when using SAX parser.

### PR DESCRIPTION
**Pull request by angelozerr**
Disable external entities when using SAX parser.

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/utils/DOMUtils.java [hunk 1]
```diff
@@ -134,4 +139,25 @@ public static DOMDocument loadDocument(String documentURI, URIResolverExtensionM
...
+	 * @throws ParserConfigurationException
+	 */
+	public static SAXParserFactory newSAXParserFactory()
+			throws SAXNotRecognizedException, SAXNotSupportedException, ParserConfigurationException {
+		SAXParserFactory factory = SAXParserFactory.newInstance();
+		// to be compliant, completely disable DOCTYPE declaration:
+		factory.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);
+		// or completely disable external entities declarations:
+		factory.setFeature("http://xml.org/sax/features/external-general-entities", false);
+		factory.setFeature("http://xml.org/sax/features/external-parameter-entities", false);
```
Comment (fbricon) [inline org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/utils/DOMUtils.java:160]: I think this is unnecessary once you disabled DOCTYPE declaration

Comment (angelozerr) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/utils/DOMUtils.java]: Ok I fixed it.

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/utils/DOMUtils.java [hunk 2]
```diff
@@ -134,4 +139,22 @@ public static DOMDocument loadDocument(String documentURI, URIResolverExtensionM
...
+	 * @return an instance of SAX parser factory by disabling external entities
+	 *         declarations.
+	 * @throws SAXNotRecognizedException
+	 * @throws SAXNotSupportedException
+	 * @throws ParserConfigurationException
+	 */
+	public static SAXParserFactory newSAXParserFactory()
+			throws SAXNotRecognizedException, SAXNotSupportedException, ParserConfigurationException {
+		SAXParserFactory factory = SAXParserFactory.newInstance();
+		// to be compliant, completely disable DOCTYPE declaration:
```
Comment (fbricon) [inline org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/utils/DOMUtils.java:156]: // to be more secure, completely ...

Comment (angelozerr) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/utils/DOMUtils.java]: fixed

**Review by fbricon — APPROVED** [review]

**Comment by angelozerr** [comment]
Thanks  @fbricon for the review

**Pull request merged**